enclave-runtime/src/lib.rs

```diff
@@ -536,13 +530,13 @@ where
 	debug!("Syncing light client!");
 	for signed_block in blocks_to_sync.into_iter() {
 		let block = signed_block.block;
-		validator.check_xt_inclusion(validator.num_relays(), &block).unwrap(); // panic can only happen if relay_id does not exist
+		let justifications = signed_block.justifications.clone();
+
+		if let Err(e) = validator_access.execute_mut_on_validator(|v| {
+			v.check_xt_inclusion(v.num_relays(), &block)?;
 
-		if let Err(e) = validator.submit_simple_header(
-			validator.num_relays(),
-			block.header().clone(),
-			signed_block.justifications.clone(),
-		) {
+			v.submit_simple_header(v.num_relays(), block.header().clone(), justifications)
+		}) {
 			error!("Block verification failed. Error : {:?}", e);
```

Comment:
That's not quite consistent. In case `v.check_xt_inclusion(v.num_relays(), &block)?;` fails, this log will be printed.

Reply:
Though we lived with an unwrap before.. so I guess we can also ignore this one.

Reply:
Also if `submit_simple_header()` fails, the log will be printed. I think I don't understand what inconsistency you mean?

Reply:
Ah, with `"Block verification failed.` you target both functions. I didn't understand `check_xt_inclusion` as a block verification. But I think you can call it that to some extent. So I agree with it, sorry for the confusion.

Reply:
Well I just left the error message as-is, even though it might not be the most descriptive anymore. What do you think would be better?

Reply:
Hm. Maybe `[Validator] Header submission failed due to ..` ?